packages/react-native-client/android/src/main/java/org/membraneframework/reactnative/MirrorableView.kt

```diff
@@ -0,0 +1,51 @@
+package org.membraneframework.reactnative
+
+import android.animation.ValueAnimator
+import android.content.Context
+import com.fishjamcloud.client.ui.VideoTextureViewRenderer
+import expo.modules.kotlin.AppContext
+import expo.modules.kotlin.views.ExpoView
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+
+abstract class MirrorableView(
+  context: Context,
+  appContext: AppContext
+) : ExpoView(context, appContext) {
+  private var mirrorVideo: Boolean = false
+  private var isFrontCamera: Boolean = false
+  var isInitialized: Boolean = false
+  val coroutineScope: CoroutineScope = CoroutineScope(Dispatchers.Main + Job())
+  abstract val fadeAnimation: ValueAnimator
+  abstract val videoView: VideoTextureViewRenderer?
+
+  fun setMirrorVideo(
+    mirrorVideo: Boolean?,
+    isFrontCamera: Boolean?
+  ) {
+    if (this.mirrorVideo == mirrorVideo && this.isFrontCamera == isFrontCamera) return
+    this.mirrorVideo = mirrorVideo ?: this.mirrorVideo
+    this.isFrontCamera = isFrontCamera ?: this.isFrontCamera
+    if (!isInitialized) return
+
+    coroutineScope.launch {
+      fadeAnimation.start()
+      delay(200)
+      videoView?.setMirror(this@MirrorableView.mirrorVideo xor this@MirrorableView.isFrontCamera)
```

Comment:
It is because:
mirror | isFront | shouldReallyMirror:
T.        | T           | F
T.        | F           | T
F.        | T           | T
F.        | F           | F
and this is exactly xor 
if you want i can add comment with documentation